3.workload.md

```diff
@@ -0,0 +1,378 @@
+# 3. Workload
+
+This section defines the workload definition schematics.
+
+Workload definitions are provided by the platform (just as traits are) so that users may inspect the platform and
+ learn what workloads are available for use. Workload definitions are not extensible to platform users (only to platform
+  operators). Thus, platform users MUST NOT be allowed to create new workload kinds.
+
+## Top-Level Attributes
+
+The following attributes are common across all schemata defined in this document. This structure is designed to allow
+ object type to be determined by scanning two fields (`apiVersion` and `kind`), and provide access to common fields
+  in the `metadata` section. The `spec` section is where type-specific information is located.
+
+Here are the attributes that provide top-level information about the workload definition.
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `apiVersion` | `string` | Y | | The specific version of the specification in use. The core types use `core.oam.dev/v1alpha2` |
+| `kind` | `string` | Y || The string `WorkloadDefinition` |
+| `metadata` | [`Metadata`](#metadata) | Y | | WorkloadDefinition metadata. |
+| `spec`| [`Spec`](#spec) | Y | | A container for the workload definition. |
+
+### Metadata
+
+The metadata describes this workload definition. The `metadata` section is defined in [the meta section of part 2](2.overview_and_terminology.md#Metadata)
+
+### Spec
+
+The specification section contains a reference to the workload schema
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `DefinitionRef` | [`DefinitionRef`](#DefinitionRef) | Y | | Workload schema reference. |
+
+
+#### DefinitionRef
+
+DefinitionRef defines where to find the workload schema
+
+| Attribute | Type | Required | Default Value | Description |
+|-----------|------|----------|---------------|-------------|
+| `name` | `string` | Y | | The workload schema's name. Must be formatted in a way that can uniquely identify a specific version of the definition  |
+
+The workload definition itself __MUST__ contain information that can be used to locate it directly. We will use that
+ information in the [Component](4.component.md) to refer to the workload schema instead of using the name of the workload definition. 
+
+We recommend that the `name` here follows the format described in [the Group, Version, Kind section of part 2](2.overview_and_terminology.md).
+
+We also recommend that the `name` of the workload (in the metadata section) to include the Group and Kind of the schema.
+
+## Workload Kinds
```

Comment:
Kinds have special meaning in K8s world. Let's name it "Workload Ladder" or sth fancy.